packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -0,0 +1,339 @@
+
+    constructor(config: AuthConfig) {
+        this.client = new ShopperLogin({
+            proxy: config.proxy,
+            parameters: {
+                clientId: config.clientId,
+                organizationId: config.organizationId,
+                shortCode: config.shortCode,
+                siteId: config.siteId
+            },
+            throwOnBadResponse: true
+        })
+
+        this.redirectURI = config.redirectURI
+        this.localStorage = this.onClient ? new LocalStorage() : new Map()
+        this.cookieStorage = this.onClient ? new CookieStorage() : new Map()
+    }
+
+    private get(name: AuthDataKeys) {
+        const storage = this.DATA_MAP[name].storage
+        const key = this.DATA_MAP[name].key
+        return storage.get(key)
+    }
+
+    private set(name: AuthDataKeys, value: string, options?: any) {
+        const key = this.DATA_MAP[name].key
+        const storage = this.DATA_MAP[name].storage
+        storage.set(key, value, options)
+        this.DATA_MAP[name].callback?.()
+    }
+
+    /**
+     * A map of the data that this auth module stores. This maps the name of the property to
+     * the storage and the key when stored in that storage. You can also pass in a "callback"
+     * function to do extra operation after a property is set.
+     */
+    private get DATA_MAP(): AuthDataMap {
+        return {
+            access_token: {
+                storage: this.localStorage,
+                key: 'access_token'
+            },
+            customer_id: {
+                storage: this.localStorage,
+                key: 'customer_id'
+            },
+            usid: {
+                storage: this.localStorage,
+                key: 'usid'
+            },
+            enc_user_id: {
+                storage: this.localStorage,
+                key: 'enc_user_id'
+            },
+            expires_in: {
+                storage: this.localStorage,
+                key: 'expires_in'
+            },
+            id_token: {
+                storage: this.localStorage,
+                key: 'id_token'
+            },
+            idp_access_token: {
+                storage: this.localStorage,
+                key: 'idp_access_token'
+            },
+            token_type: {
+                storage: this.localStorage,
+                key: 'token_type'
+            },
+            refresh_token_guest: {
+                storage: this.cookieStorage,
+                key: 'cc-nx-g',
+                callback: () => {
+                    this.cookieStorage.delete('cc-nx')
+                }
+            },
+            refresh_token_registered: {
+                storage: this.cookieStorage,
+                key: 'cc-nx',
+                callback: () => {
+                    this.cookieStorage.delete('cc-nx-g')
+                }
+            }
+        }
+    }
+
+    /**
+     * Every promise method in this class returns the same data via this method.
```

Comment:
```suggestion
     * Every method in this class that returns a `TokenResponse` constructs it via this getter.
```
Just a bit more clear.